api/src/main/java/io/minio/MinioClient.java

```diff
@@ -1620,12 +1620,10 @@ public InputStream getObject(String bucketName, String objectName, long offset,
     }
 
     Map<String,String> headerMap = new HashMap<>();
-    if (offset > 0) {
-      if (length != null) {
-        headerMap.put("Range", "bytes=" + offset + "-" + (offset + length - 1));
-      } else {
-        headerMap.put("Range", "bytes=" + offset + "-");
-      }
+    if (length != null) {
```

Comment:
`if (offset > 0)` has to be there else wrong range will be sent for negative values.

Reply:
The linter was complaining in CI because this check is actually unnecessary; if you look a few lines up, there is already a check for the case when offset < 0. I had to remove the duplicate check here in order to satisfy the linter.

Reply:
got it